app/mailers/user_mailer.rb

```diff
@@ -1,8 +1,11 @@
 class UserMailer < ApplicationMailer
-  def authentication
-    @authentication = params[:authentication]
-    @user = @authentication.user
+  before_action do
+    @user = params[:user] || Current.user
+  end
+
+  def authentication(authentication = nil)
+    @authentication = authentication || @user.authentications.last
```

Comment:
not sure if being implicit here is the best move, seems to be on the line - @garyhtou?